websocket/websocket.go

```diff
@@ -0,0 +1,181 @@
+package websocket
+
+import (
+	"errors"
+	"log"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// Client for forex api instance
+type Client struct {
+	Address   *string
+	Query     *string
+	Conn      *websocket.Conn
+	Listeners *[]func(interface{})
+	logger    *log.Logger
+}
+
+var reconnectEvents = []int{
+	websocket.CloseGoingAway,
+	websocket.CloseAbnormalClosure,
+}
+
+var reconnectIntervalSeconds int64 = 5
+
+// New : create the weboscket instance
+func New(address *string, logger *log.Logger) (*Client, error) {
+	if address == nil {
+		return nil, errors.New("'address' is missing")
+	}
+	if logger == nil {
+		logger = log.New(os.Stderr, "WebSocket: ", log.LstdFlags)
+	}
+	return &Client{
+		Address:   address,
+		logger:    logger,
+		Listeners: &[]func(interface{}){},
+	}, nil
+}
+
+// Connect to websocket endpoint
+func (client *Client) Connect() error {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	// connection
+	u := url.URL{
```

Comment:
Faced a problem here, I'm trying to mock forex service in test, I need to pass this url, `ws://localhost:4444/ws/forex`, the problem is I cannot set path and if I pass the whole address its encoded. Also I cannot set wss, too.
If you want to use url type, take this as an argument to `New`

Reply:
updated. now you can pass full url likes `ws://localhost:4444/ws/forex`. but the query string will be placed anyway.